driver/modern_bpf/programs/tail_called/events/syscall_dispatched_events/getsockopt.bpf.c

```diff
@@ -68,9 +68,15 @@ int BPF_PROG(getsockopt_x,
 	u8 optname = (u8)extract__syscall_argument(regs, 2);
 	auxmap__store_u8_param(auxmap, sockopt_optname_to_scap(level, optname));
 
+	/* `optval` and `optlen` will be the ones provided by the user if the syscall fails
+	 * otherwise they will be directly extracted from the socket.
+	 */
+
 	/* Parameter 5: optval (type: PT_DYN) */
 	unsigned long optval = extract__syscall_argument(regs, 3);
-	u16 optlen = (u16)extract__syscall_argument(regs, 4);
+	u16 optlen = 0;
```

Comment:
Based on the syscall definition:
````c
SYSCALL_DEFINE5(getsockopt, int, fd, int, level, int, optname,
                char __user *, optval, int __user *, optlen)
```
`u16` should be an `int`.

So far I could `grep` `socklen_t` and `__socklen_t` are 32-bit wide in size.

Reply:
Other than that, this also looks good to me.

Reply:
great catch!

Reply:
Fixed, thanks